benchmarking/smartseq2/picard_qc_analysis/QC_metrics_analysis.R

```diff
@@ -0,0 +1,387 @@
+# R script to do metrics test
+# loading library
+library(ggplot2)
+library(MASS)
+library(plyr)
+library(reshape2)
+library(ggpubr)
+library(gridExtra)
+library(ggpmisc)
+library(cowplot)
+library(optparse)
+
+##convert p-value to star char
+convert2star <- function(x) {
+  if (x > 0.05) {
+    p <- 'ns'
+  } else if (x > 0.01) {
+    p <- '*'
+  } else if (x > 0.001) {
+    p <- '**'
+  } else if (x > 0.0001) {
+    p <- '***'
+  } else{
+    p <- '****'
+  }
+  return(p)
+}
+## extra theme
+addTheme <- function(p) {
+  p <- p + theme(legend.position = "top")
+  p <-
+    p + theme(plot.title = element_text(
+      hjust = 0.5,
+      size = 20,
+      face = 'bold'
+    ))
+  p <-
+    p + theme(
+      axis.title.x = element_text(
+        color = 'black',
+        size = 18,
+        face = 'bold'
+      ),
+      axis.title.y = element_text(
+        size = 18,
+        color = 'black',
+        face = 'bold'
+      )
+    )
+  p <-
+    p + theme(
+      axis.text.y = element_text(
+        color = 'black',
+        size = 18,
+        face = 'bold'
+      ),
+      axis.text.x = element_text(
+        color = 'black',
+        size = 18,
+        face = 'bold'
+      )
+    )
+  p <-
+    p + theme(legend.text = element_text(size = 15, face = 'bold'),
+              legend.title = element_blank())
+  return(p)
+}
+# lm test
+# scatter plot and fitted line
+# also plot 1x1 line
+plotlm <- function(df) {
+  # lm test
+  x <- df$Base
+  y <- df$Updated
+  fit <- lm(y ~ x)
+  # summary
+  sfit <- summary(fit)
+  # r2
+  r2  <- round(sfit$adj.r.squared, 3)
+  # intercept and slope
+  beta <- round(sfit$coefficients[2], 3)
+  a <- round(sfit$coefficients[1], 3)
+  # f-stats and p values
+  f <- sfit$fstatistic
+  fpval <- pf(f[1], f[2], f[3], lower.tail = F)
+  fpval.s <- convert2star(fpval)
+  # start to plot
+  coefs <-
+    data.frame(
+      'ic' = c(0, a),
+      's' = c(1, beta),
+      'tl' = c('1x1', 'fitted')
+    )
+  my.formula <- y ~ x
+  p <-
+    ggplot(data = df, aes(x = Base, y = Updated)) + geom_point(shape = 1) + theme_bw(base_size = 20)
+  p <- p + stat_poly_eq(
+    formula = my.formula,
+    eq.with.lhs = "italic(hat(y))~`=`~",
+    aes(
+      label = paste(..eq.label.., ..rr.label.., sep = "~~~"),
+      size = 8
+    ),
+    parse = TRUE,
+    size = 8
+  )
+  p <-
+    p + stat_fit_glance(
+      method = 'lm',
+      method.args = list(formula = my.formula),
+      geom = 'text',
+      aes(
+        label = paste("P-value:", fpval.s , sep = ""),
+        size = 12
+      ),
+      label.x.npc = 'left',
+      label.y.npc = 0.85,
+      size = 8
+    )
+  p <- p + theme(legend.position = "top")
+  p <- addTheme(p)
+  p <- p + xlab(paste('Base')) + ylab(paste('Updated'))
+  p <-
+    p + geom_abline(
```

Comment:
the smooth should not change so much in our case since x~y are highly linear correlated. so even after "smoothed", most data points are still lined up closely.

Reply:
I just realized what you asked for. Yes if we only want to plot one line I can use `geom_smooth`. What I was trying to do is to plot two lines, one is the fitted regression line and another is 1-1 line. I want to see how far the fitted line off to 1-1 line.